### ISSUE TITLE
wdltool should report optionals (even with defaults)

### ISSUE THREAD

**Issue by cjllanwarne** [broadinstitute/cromwell#2532]
When wdltool generates an inputs JSON, you want to include all required and optional values, even if the have defaults. For example:

```
workflow foo {
  Int a
  Int b = 5
  Int? c
  Int? d = 5
  Int e = a + b
  Int? f = a + b # Not sure how to handle this one
}
```

Should create inputs, something like this (not 100% sure of the format it creates):
```
{
  "foo.a": "Int"
  "foo.c": "Int?"
  "foo.d": "Int? with default"
  "foo.f": "Int? with default"
}
```

The handling of `e` and `f` is probably why this hasn't been

**Comment by LeeTL1220** [comment]
@cjllanwarne Yes, my money is that you have pegged this exactly.  I would love this update.  I prefer the ``Int? mem=4`` for specifying default values.  Otherwise, we have to pepper our command and runtime blocks with ``${default=4 mem}`` or, even worse, something with ``select_first``

We often have inputs that are derived (e.g. ``e``) and we do not want these exposed in ``wdltool inputs ...``.

I do not have a good idea for how to handle ``f``.  I'm assuming you do not have access to the raw expression when rendering ``wdltool inputs ...``, so can you just say that it has a complex default expression?

**Comment by Redmar-van-den-Berg** [comment]
Not to hijack this issue, but would it be possible to put the actual default values in the .json file? Right now, users can change an optional value, but it is very hard to figure out what the default value is. Simply putting it in the json file would save users a lot of hunting through .wdl files to figure out what a sensible value is.

**Comment by geoffjentry** [comment]
@cjllanwarne what's the current situation here? This is an old enough ticket that it might just work now
